hail/python/hail/experimental/import_gtf.py

```diff
@@ -0,0 +1,132 @@
+       attribute field of the GTF file.
+
+       .. note::
+
+           The "end" field in the table will be incremented by 1 in
+           comparison to the value found in the GTF file, as the end
+           coordinate in a GTF file is inclusive while the end
+           coordinate in Hail is exclusive.
+
+       Example
+       -------
+
+       >>> ht = hl.experimental.import_gtf('data/test.gtf', key='gene_id')
+       >>> ht.describe()
+
+       .. code-block:: text
+
+           ----------------------------------------
+           Global fields:
+           None
+           ----------------------------------------
+           Row fields:
+               'seqname': str
+               'source': str
+               'feature': str
+               'start': int32
+               'end': int32
+               'score': float64
+               'strand': str
+               'frame': int32
+               'havana_gene': str
+               'exon_id': str
+               'havana_transcript': str
+               'transcript_name': str
+               'gene_type': str
+               'tag': str
+               'transcript_status': str
+               'exon_number': str
+               'level': str
+               'transcript_id': str
+               'transcript_type': str
+               'gene_id': str
+               'gene_name': str
+               'gene_status': str
+           ----------------------------------------
+           Key: ['gene_id']
+           ----------------------------------------
+
+       Parameters
+       ----------
+
+       path : :obj:`str`
+           File to import.
+       key : :obj:`str` or :obj:`list` of :obj:`str`
+           Key field(s). Can be tag name(s) found in the attribute field
+           of the GTF file.
+
+       Returns
+       -------
+       :class:`.Table`
+       """
+
+    ht = hl.import_table(path,
+                         comment='#',
+                         no_header=True,
+                         types={'f3': hl.tint,
+                                'f4': hl.tint,
+                                'f5': hl.tfloat,
+                                'f7': hl.tint},
+                         missing='.',
+                         delimiter='\t')
+
+    ht = ht.rename({'f0': 'seqname',
+                    'f1': 'source',
+                    'f2': 'feature',
+                    'f3': 'start',
+                    'f4': 'end',
+                    'f5': 'score',
+                    'f6': 'strand',
+                    'f7': 'frame',
+                    'f8': 'attribute'})
+
+    ht = ht.annotate(end=ht['end'] + 1)
+    ht = ht.annotate(attribute=hl.dict(
+        hl.map(lambda x: (x.split(' ')[0],
+                          x.split(' ')[1].replace('"', '').replace(';$', '')),
+               ht['attribute'].split('; '))))
+
+    attributes = list(ht.aggregate(
+        hl.set(hl.flatten(hl.agg.collect(ht['attribute'].keys())))))
```

Comment:
I think you can probably do this without collecting the full `rows x keys` list. Something like

```python
attributes = ht.aggregate(hl.agg.collect_as_set(hl.agg.explode(ht['attribute'].keys())))
```